lib/avalon/audit_objects.rb

```diff
@@ -16,20 +16,12 @@ def self.error_stats(bad_objects)
       # Total errors
       stats = [ { heading: I18n.t('audit.total_header_html').html_safe,
                   value: bad_objects.count } ]
-      object_class_counts = {}
-      collection_counts = {}
 
-      bad_objects.each do |object|
-        # Count occurences for each object class
-        object_class_counts[object.class] ||= 0
-        object_class_counts[object.class] += 1
+      object_class_counts = bad_objects.group_by(&:class).transform_values! {|v| v.length}
+      collection_counts = bad_objects.select { |obj| obj.respond_to?(:collection) }
+                            .group_by {|obj| obj.collection.name }
+                            .transform_values! {|v| v.length }
```

Comment:
Sanity checking that I didn't just create fun new problems: Is Avalon running Ruby > 2.4 in Prod? It turns out `transform_values!` is a fairly recent addition.

(if not `.map {|k, v| {k => v.length} }` is the compatible replacement, I guess)

Reply:
Should be okay. Current Avalon 5 is ruby 2.3, but Avalon 6 docker uses ruby 2.4.4, so we can request that for the upgrade. (@weiweishi : is there a place such decisions can be documented for ITS?)

Reply:
Ok cool